src/plugins/layout/shadowdom.ts

```diff
@@ -240,29 +214,18 @@ export class ShadowDom {
     };
 
     // Collect all new nodes in the top-down order
-    let topNewNodes = this.doc.getElementsByClassName(NewNodeClassName);
-    while (topNewNodes.length > 0) {
-      let topNode = topNewNodes[0] as IShadowDomNode;
-      let discoverQueue: IShadowDomNode[] = [ topNode ];
-      while (discoverQueue.length > 0) {
-        let next = discoverQueue.shift();
-        if (this.hasClass(next, NewNodeClassName)) {
-          summary.newNodes.push(next);
-          this.removeClass(next, NewNodeClassName);
-          // Add children right-to-left
-          let children = next.childNodes;
-          for (let i = children.length - 1; i >= 0; i--) {
-            discoverQueue.push(children[i] as IShadowDomNode);
-          }
-        }
-      }
+    let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
```

Comment:
Definitely a big work item! We tried to avoid in-code TODOs, but we can create a tracking item on github

---
In reply to: [132227437](https://github.com/Microsoft/clarity-js/pull/25#discussion_r132227437) [](ancestors = 132227437)